src/lib/mina_base/account_update.ml

```diff
@@ -1570,7 +1628,9 @@ let verification_key_update_to_option (t : t) :
   Zkapp_basic.Set_or_keep.map ~f:Option.some t.body.update.verification_key
 
 let of_fee_payer ({ body; authorization } : Fee_payer.t) : t =
-  { authorization = Signature authorization; body = Body.of_fee_payer body }
+  { authorization = Fee_payer.Authorization.to_control authorization
+  ; body = Body.of_fee_payer body
```

Comment:
We need to update `authorization_kind` in the body based on the actual authorization, right?

Reply:
It should already be correctly set in the original `Fee_payer.t`.

Reply:
I don't think so? It is always being set to `Signature` https://github.com/MinaProtocol/mina/blob/3a1a78d64ae811ac407b4d580e31a37e34ecda98/src/lib/mina_base/account_update.ml#L1155